File: backend/src/main/java/com/zihler/wiki/domain/values/WikiPages.java
package com.zihler.wiki.domain.values;

import com.zihler.wiki.domain.entity.WikiPage;

import java.util.HashSet;
import java.util.Set;

public class WikiPages {
    private Set<WikiPage> wikiPages;

    private WikiPages(Set<WikiPage> wikiPages) {
        this.wikiPages = new HashSet<>();
        this.wikiPages = wikiPages;
    }

    public static WikiPages from(Set<WikiPage> wikiPages) {
        return new WikiPages(wikiPages);
    }

    public Set<WikiPage> getWikiPages() {
        return wikiPages;
    }

}

File: frontend/src/app/wiki/application.use_case/wiki-pages.ts
import {WikiPage} from "../domain/wiki-page";

export class WikiPages {

  static from(response): WikiPage[] {
    let wikiPages: WikiPage[] = [];
    response.wikiPages.forEach(wikiPage => {
      wikiPages.push(WikiPage.from(wikiPage));
    });
    return wikiPages;
  }
}
